share/man/man7/c.7

```diff
@@ -211,17 +212,118 @@ It incorporates the Technical Corrigendum 1 (ISO/IEC 9899:2011/COR1:2012)
 which was published in 2012.
 It addressed defects and deficiencies in C11 without introducing new features,
 only corrections and clarifications.
-Since there were no major changes in C17, the current standard for
-Programming Language C, is still considered C11 \(em ISO/IEC 9899:2011, published
-2011-12-08.
 .Pp
-The next standard, the fifth, is currently referred to as C2x and is scheduled
-to be adopted by the end of 2021, with a publication date of 2022.
-When published, it will cancel and replace the fourth edition, ISO/IEC
-9899:2018.
+C23, formally ISO/IEC 9899:2024, is the current standard with significant
+updates that supersede C17 (ISO/IEC 9899:2018).
+The standardization effort began in 2016, informally as C2x, with the first
+WG14 meeting in 2019, and was officially published on October 31, 2024.
+C23 was originally anticipated for an earlier release, but the timeline was
+extended due to COVID-19 pandemic.
+With C23, the value of __STDC_VERSION__ has been updated from 201710L to
+202311L.
+Key changes include (but are not limited to) the following:
+.Bl -bullet -offset indent
+.It
+Add null pointer type nullptr_t and the nullptr keyword
+.It
+Add constexpr keyword as a storage-class specifier for objects
+.It
+Redefine the usage of the auto keyword to support type inference while also
+retaining its previous functionality as a storage-class specifier when used
+with a type
+.It
+Add %b binary conversion specifier to the printf() and scanf() function
```

Comment:
To nitpick, `mandoc -T lint` gives STYLE notices for all these function names without `.Fn`.

Normally I wouldn't bother mentioning style issues, but these are the only issues with `c.7`; previously, there were no issues at all with this file.

Reply:
Right, that’s why I mentioned the focus herein is only on adding information for the recent C23 release and the overall structure and macro usage in the man page shall be improved (and not just the function names—there’s quite a bit that needs proper macro usage, I might add.) Landing the update first would be ideal, and I can work on the style next week in a separate PR.

Reply:
Forget it, I'll do the initial revision in an hour.